File: pset6/cash/cash.py
QUARTER = 25
DIME = 10
NICKEL = 5


def main():
    dollars = get_positive()
    cents = calculate_cents(dollars)
    coins = calculate_minimum_coins(cents)
    print(coins)
    
    
def get_positive():
    while True:
        try:
            value = float(input("change owed: "))
            if value > 0:
                return value
        except ValueError:
            continue
        
        
def calculate_cents(dollars):
    return round(dollars * 100)
    
    
def calculate_minimum_coins(cents):
    coins = 0
    
    coins, cents = calculate_coins_value(coins, cents, QUARTER)
    coins, cents = calculate_coins_value(coins, cents, DIME)
    coins, cents = calculate_coins_value(coins, cents, NICKEL)
    coins += cents
    
    return coins
    
    
def calculate_coins_value(coins, cents, value):
    if (cents >= value):
        coins += cents // value
        cents = cents % value
        
    return coins, cents
    
    
main()

File: pest2/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

float get_positive(void);
int cauculate_minimum_coins(int cents);

int main(void)
{
    float dollars = get_positive();
    int cents = round(dollars * 100);
    int coins = cauculate_minimum_coins(cents);
    printf("%d\n", coins);
}

float get_positive(void)
{
    float positive;
    
    do
    {
        positive = get_float("change owed: ");
    }
    while (positive < - 0);
    
    return positive;
}

int cauculate_minimum_coins(int cents)
{
    int coins = 0;

    if (cents >= 25)
    {
        coins += cents / 25;
        cents = cents % 25;
    }
    
    if (cents >= 10)
    {
        coins += cents / 10;
        cents = cents % 10;
    }
    
    if (cents >= 5)
    {
        coins += cents / 5;
        cents = cents % 5;
    }
    
    coins += cents;
    
    return coins;
}

